yarn-project/archiver/src/archiver/eth_log_handlers.ts

```diff
@@ -107,7 +107,7 @@ async function getBlockFromCallData(
   if (functionName !== 'process') {
     throw new Error(`Unexpected method called ${functionName}`);
   }
-  const [headerHex, archiveRootHex, bodyHex] = args! as [Hex, Hex, Hex, Hex];
+  const [headerHex, archiveRootHex, , bodyHex] = args! as [Hex, Hex, Hex, Hex, Hex];
```

Comment:
When you are working on #3938 would it not mess with your decoding here? It would not longer be practical to get the full block from just this kind of function call as the `bodyHex` would not be here. 

This part seems to be very reliant on the content being published at the same time for getting any data. 

In my mind you would need to fetch the body separately through the extra event you added. And then as l2 blocks are seen you can combine the two pieces together for the block. 

As this is the synching part, it can be handled separately, a new pr for it might be nice to have the split fully dealt with? This seems a bit like cheating. But let us talk about it, as the new changes for messages would likely mess with it a bit, so an initial pr could focus on just needing to pass the messages and nothing else. That way we can address synching separately to the hashing updates.